components/feature/session-bundling/src/main/java/mozilla/components/feature/session/bundling/SessionBundleStorage.kt

```diff
@@ -0,0 +1,147 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.session.bundling
+
+import android.arch.lifecycle.LiveData
+import android.arch.lifecycle.Transformations
+import android.arch.paging.DataSource
+import android.content.Context
+import android.support.annotation.CheckResult
+import mozilla.components.browser.session.Session
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.browser.session.storage.AutoSave
+import mozilla.components.feature.session.bundling.db.BundleDatabase
+import mozilla.components.feature.session.bundling.db.BundleEntity
+import mozilla.components.feature.session.bundling.ext.toBundleEntity
+import java.util.concurrent.TimeUnit
+
+/**
+ * A [Session] storage implementation that saves snapshots as a [SessionBundle].
+ *
+ * @param bundleLifetime The lifetime of a bundle controls whether a bundle will be restored or whether this bundle is
+ * considered expired and a new bundle will be used.
+ */
+class SessionBundleStorage(
+    context: Context,
+    private val bundleLifetime: Pair<Long, TimeUnit>
```

Comment:
Passing a `bundeLifetime` to the constructor will have all bundles have the same lifetime. I am wondering if we should instead add a `createBundle` call to this class. Then apps / or other components could create bundles either based on different times/intervals (~3pm, this afternoon) or even based on topics etc. I might be getting ahead of myself, but it seems a method to create new bundles might be better? Basically the `SessionBundleStorage` would just expose method to manage bundles. Then it's up to the app / other components to define when that happens based on whatever algorithm they want to use.

Reply:
This will make `restore` more complicated though as we'd need to specify what `latest` means, but it seems doable as long as we still sort by `saved_at`.

Reply:
Yeah, currently it's not very flexible. I was thinking of turning this into some kind of builder-pattern-like thing that you can configure to create bundles the way you like (or customize). However that's something I would like to do in a future iteration.

Reply:
OK, sounds good!